### ISSUE TITLE
1.8-RC2 - tz_offset yields odd results

### ISSUE THREAD

**Issue by teckel12** [iNavFlight/inav#2409]
I set the `tz_offset` value to -240 (4 hours).  The OSD was off by 16 minutes?  So, I set `tz_offest` to -256 and now the time is correct.  Something isn't right here.

Also, the file timestamp on my blackbox logs were still UTC time.

**Comment by digitalentity** [comment]
Minutes offset here is unsigned which causes issues. @teckel12 in your case time is probably off by 1088 hours and 16 minutes. @fiam can you look into this?
```
static void dateTimeWithOffset(dateTime_t *dateTimeOffset, dateTime_t *dateTimeInitial, uint16_t minutes)
{
    rtcTime_t initialTime = dateTimeToRtcTime(dateTimeInitial);
    rtcTime_t offsetTime = rtcTimeMake(rtcTimeGetSeconds(&initialTime) + minutes * 60, rtcTimeGetMillis(&initialTime));
    rtcTimeToDateTime(dateTimeOffset, offsetTime);
}
```

**Comment by teckel12** [comment]
Sorry that I live "left" of UTC. On this side of the planet, we refer to our time zone as -4 or -5 hours from UTC (depending on DST), at least on the east coast. I can understand how "negative time" would seem unlogical. But that's how we refer to our time zone vs. UTC.

**Comment by digitalentity** [comment]
Fixed by #2413

**Comment by teckel12** [comment]
Those of us this side of the world thank you.
